### PR TITLE
Add consent banner component

### LINKED ISSUE #1398: Create cookie & privacy banner component
**Is your feature request related to a problem? Please describe.**
We need a banner that can be used as the privacy and cookie banner on canonical and the cookie banner on AMP.

The privacy banner:
Screen Shot privacy banner

The cookie banner:
Screen Shot cookie banner

**Describe the solution you'd like**
- Implement the component, matching the styles of the existing Privacy / Cookie Banner. We will not include the gear/tick icons in this MVP version. 
- AMP/Canonical conditional logic should be in the Consent Banner Container, added in a separate PR. 
- Do not integrate into the Article container in this ticket. 
- See CookieBanner component for Canonical proof of concept: https://github.com/bbc/simorgh/pull/1336/files#diff-d51fea290710062195af8293a7eb7f13R1
See ConsentBanner component for AMP proof of concept: https://github.com/bbc/simorgh/pull/1306/files#diff-fb8f87665273f036db9aa5025de2a314 
Cookie 

Exports from the component:
ConsentBannerButton component (e.g. with hover/focus styles)
ConsentBannerLink component (e.g. with hover/focus styles)
ConsentBanner component
ConsentBannerText component (we could generate a block with attributes like the Optimo Fragments, however we prefer for this MVP version to use three hard-coded strings e.g. text1, text2, text2Url, text3)

We need the component to be able to be used in four scenarios for AMP and Canonical, defined in these issues #1399 #1400

**Describe alternatives you've considered**
N/A

**Testing notes**
Regression test the Privacy & Cookie banner component stories to make sure that the styles within Storybook match those that are currently within Orbit.

Dev insight: Jest snapshot tests should be sufficient, since this is creating the component and not integrating it into the article. We should have four storybook stories - for the Privacy banner and for the cookie banner, for both AMP and Canonical. 

**Additional context**

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by dr3**
~~Blocked on https://github.com/bbc/simorgh/pull/1394 using the new psammead-styles and fontface~~ DONE

Resolves #1398

**Overall change:** Adds consent banner that exports 4 things, the banner itself, the button, a link and a text component 

**Code changes:**

- Adds component
- Adds example in storybook

### MVP Styles
This component uses a mixture of CSS reused from orbit (the original bbc implementation), and some psammead constants where possible. It is an MVP implementation and is intended to be redone before it is moved to Psammead after V1.0

**Whats left to do for the component?**
We need to add the icons as well as clean up styles. These styles can likely be optimised and modernised using grid

---

image

image

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [x] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/components/ConsentBanner/index.jsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
+    padding: ${GEL_MARGIN_ABOVE_400PX};
+  }
+`;
+
+const CenterWrapper = styled.div`
+  max-width: 1280px;
+  margin: 0 auto;
+
+  &::after {
```
Comment (dr3) [inline src/app/components/ConsentBanner/index.jsx:46]: This is here to ensure that the background colour covers the banner as the inner elements are float. The alernative is to have another div inside, but this is just copied from orbit.

File: src/app/components/ConsentBanner/index.jsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+const Options = styled.ul`
+  ${GEL_LONG_PRIMER}
+  color: ${ORANGE};
+  font-weight: 600;
+  padding: 0;
+  margin: 0;
+  list-style-type: none;
+
+  & li + li {
```
Comment (dr3) [inline src/app/components/ConsentBanner/index.jsx:81]: Style every `li` except the first

File: src/app/components/ConsentBanner/index.jsx (hunk 1 above)
Comment (ChrisBAshton) [reply src/app/components/ConsentBanner/index.jsx:46]: Can you put this in an inline comment above the block?

File: src/app/components/ConsentBanner/index.jsx (hunk 2 above)
Comment (ChrisBAshton) [reply src/app/components/ConsentBanner/index.jsx:81]: Please put in an inline comment above the block

File: src/app/components/ConsentBanner/index.jsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    width: 53%;
+  }
+
+  a {
+    color: ${C_CONSENT_ACTION};
+    text-decoration: none;
+  }
+
+  a:focus,
+  a:hover {
```
Comment (ChrisBAshton) [inline src/app/components/ConsentBanner/index.jsx]: Can we move this inside the block starting line 90, like so:

```scss
a {
  ... etc
  
  &:focus, &:hover {
    ...etc
  }
}
```

File: src/app/components/ConsentBanner/index.jsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  background: none;
+  border: none;
+  padding: 0;
+  margin: 0;
+  cursor: pointer;
+
+  &:focus,
+  &:hover {
+    color: ${C_WHITE};
+  }
```
Comment (ChrisBAshton) [inline src/app/components/ConsentBanner/index.jsx]: Actually this block is used a few times throughout this file - can we put it in one place and import it?

```js
const hoverFocusStyles = `
  &:focus, &:hover {
    color: ${C_WHITE};
  }
`;

etc
```

File: src/app/components/ConsentBanner/index.stories.jsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  ConsentBannerButton,
+  ConsentBannerLink,
+  ConsentBannerText,
+} from '.';
+
+const Accept = <ConsentBannerButton onClick={() => {}}>OK</ConsentBannerButton>;
+
+const Reject = (
+  <ConsentBannerLink href="https://hello.com" onClick={() => {}}>
+    Find out whats changed
```
Comment (ChrisBAshton) [inline src/app/components/ConsentBanner/index.stories.jsx]:
```suggestion
    Find out what's changed
```

File: src/app/components/ConsentBanner/index.stories.jsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  ConsentBanner,
+  ConsentBannerButton,
+  ConsentBannerLink,
+  ConsentBannerText,
+} from '.';
+
+const Accept = <ConsentBannerButton onClick={() => {}}>OK</ConsentBannerButton>;
+
+const Reject = (
+  <ConsentBannerLink href="https://hello.com" onClick={() => {}}>
```
Comment (ChrisBAshton) [inline src/app/components/ConsentBanner/index.stories.jsx]:
```suggestion
  <ConsentBannerLink href="https://www.bbc.co.uk/usingthebbc/your-data-matters">
```

What's the `onClick` for? Don't think we should have this for links (`Find out more`), only buttons (`OK`)

File: src/app/components/ConsentBanner/index.test.jsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    'should correctly render',
+    <ConsentBannerButton onClick={() => {}}>OK</ConsentBannerButton>,
+  );
+});
+
+describe('ConsentBannerLink', () => {
+  shouldMatchSnapshot(
+    'should correctly render',
+    <ConsentBannerLink href="https://hello.com" onClick={() => {}}>
+      Find out whats changed
```
Comment (ChrisBAshton) [inline src/app/components/ConsentBanner/index.test.jsx]:
```suggestion
      Find out what's changed
```

File: src/app/components/ConsentBanner/index.test.jsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  shouldMatchSnapshot(
+    'should correctly render',
+    <ConsentBannerButton onClick={() => {}}>OK</ConsentBannerButton>,
+  );
+});
+
+describe('ConsentBannerLink', () => {
+  shouldMatchSnapshot(
+    'should correctly render',
+    <ConsentBannerLink href="https://hello.com" onClick={() => {}}>
```
Comment (ChrisBAshton) [inline src/app/components/ConsentBanner/index.test.jsx]:
```suggestion
    <ConsentBannerLink href="https://www.bbc.co.uk/usingthebbc/your-data-matters">
```

**Review by ChrisBAshton — CHANGES_REQUESTED** [review]
> It is an MVP implementation and is intended to be redone before it is moved to Psammead after V1.0

Can you update the description to say what's left to do? AFAIK it's "translations" (handled when we integrate) and "icons", as well as triggering whatever the `onClick` should do for "OK".

File: src/app/components/ConsentBanner/index.jsx (hunk 3 above)
Comment (dr3) [reply src/app/components/ConsentBanner/index.jsx]: done :)

File: src/app/components/ConsentBanner/index.jsx (hunk 4 above)
Comment (dr3) [reply src/app/components/ConsentBanner/index.jsx]: done :)

File: src/app/components/ConsentBanner/index.jsx (hunk 2 above)
Comment (dr3) [reply src/app/components/ConsentBanner/index.jsx:81]: done :)

File: src/app/components/ConsentBanner/index.jsx (hunk 1 above)
Comment (dr3) [reply src/app/components/ConsentBanner/index.jsx:46]: done :)

**Comment by dr3** [comment]
> Can you update the description to say what's left to do? AFAIK it's "translations" (handled when we integrate) and "icons", as well as triggering whatever the `onClick` should do for "OK".

Ive added a description about whats left for the component, the translations and onclicks should come with the container and its optimisations :)

File: src/app/components/ConsentBanner/__snapshots__/index.test.jsx.snap [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+    font-size: 1.25rem;
+    line-height: 1.5rem;
+  }
+}
+
+@media (min-width:37.5em) {
+  .c2 {
+    width: 22%;
+    margin-right: 3.5%;
+    float: left;
```
Comment (pjlee11) [inline src/app/components/ConsentBanner/__snapshots__/index.test.jsx.snap:64]: ew.

File: src/app/components/ConsentBanner/__snapshots__/index.test.jsx.snap [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+@media (min-width:37.5em) {
+  .c3 {
+    font-size: 0.875rem;
+  }
+}
+
+@media (min-width:37.5em) {
+  .c3 {
+    width: 18%;
+    float: right;
```
Comment (pjlee11) [inline src/app/components/ConsentBanner/__snapshots__/index.test.jsx.snap:77]: ew.

File: src/app/components/ConsentBanner/index.jsx [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+} from '@bbc/gel-foundations/breakpoints';
+import {
+  GEL_MARGIN_BELOW_400PX,
+  GEL_MARGIN_ABOVE_400PX,
+} from '@bbc/gel-foundations/spacings';
+
+const Wrapper = styled.div`
+  font-family: ${GEL_FF_REITH_SANS};
+  background-color: ${C_CONSENT_BACKGROUND};
+  padding: 16px ${GEL_MARGIN_BELOW_400PX};
```
Comment (pjlee11) [inline src/app/components/ConsentBanner/index.jsx]:
```suggestion
  padding: 1rem ${GEL_MARGIN_BELOW_400PX};
```

File: src/app/components/ConsentBanner/index.jsx [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+  C_WHITE,
+} from '@bbc/psammead-styles/colours';
+import {
+  GEL_GREAT_PRIMER,
+  GEL_LONG_PRIMER,
+  GEL_FF_REITH_SANS,
+} from '@bbc/gel-foundations/typography';
+import {
+  GEL_GROUP_3_SCREEN_WIDTH_MIN,
+  GEL_GROUP_2_SCREEN_WIDTH_MIN,
```
Comment (pjlee11) [inline src/app/components/ConsentBanner/index.jsx:17]:
```suggestion
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_5_SCREEN_WIDTH_MIN,
```

File: src/app/components/ConsentBanner/index.jsx (hunk 11 above)
Comment (pjlee11) [reply src/app/components/ConsentBanner/index.jsx]:
```suggestion
  padding: ${GEL_SPACING_DBL} ${GEL_MARGIN_BELOW_400PX};
```

Better yet use the const.

File: src/app/components/ConsentBanner/index.jsx [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+  }
+`;
+
+/*
+ * The '&::after' below is to ensure that the background colour covers the
+ * banner as the inner elements are float. The alernative is to have
+ * another div inside. This implementation mirrors the current orbit banner.
+ */
+const CenterWrapper = styled.div`
+  max-width: 1280px;
```
Comment (pjlee11) [inline src/app/components/ConsentBanner/index.jsx]:
```suggestion
  max-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN};
```

File: src/app/components/ConsentBanner/index.stories.jsx [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+
+const Reject = (
+  <ConsentBannerLink href="https://www.bbc.co.uk/usingthebbc/your-data-matters">
+    Find out what&apos;s changed
+  </ConsentBannerLink>
+);
+
+const Text = (
+  <ConsentBannerText>
+    We have made some <a href="https://www.bbc.com/news">important</a> changes
```
Comment (ChrisBAshton) [inline src/app/components/ConsentBanner/index.stories.jsx]: This link is weird 🤔

File: src/app/components/ConsentBanner/index.stories.jsx [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+  ConsentBannerButton,
+  ConsentBannerLink,
+  ConsentBannerText,
+} from '.';
+
+const Accept = <ConsentBannerButton onClick={() => {}}>OK</ConsentBannerButton>;
+
+const Reject = (
+  <ConsentBannerLink href="https://www.bbc.co.uk/usingthebbc/your-data-matters">
+    Find out what&apos;s changed
```
Comment (ChrisBAshton) [inline src/app/components/ConsentBanner/index.stories.jsx:14]: 😬 do we have to encode the apostrophe?

File: src/app/components/ConsentBanner/index.jsx [hunk 16]
```diff
@@ -0,0 +1,151 @@
...
+const Options = styled.ul`
+  ${GEL_LONG_PRIMER}
+  color: ${C_CONSENT_ACTION};
+  font-weight: 600;
+  padding: 0;
+  margin: 0;
+  list-style-type: none;
+
+  & li + li {
+    padding-top: 8px;
```
Comment (pjlee11) [inline src/app/components/ConsentBanner/index.jsx]:
```suggestion
    padding-top: ${GEL_SPACING};
```

**Review by ChrisBAshton — DISMISSED** [review]
Looks ok to me - thanks!

File: src/app/components/ConsentBanner/index.stories.jsx (hunk 15 above)
Comment (dr3) [reply src/app/components/ConsentBanner/index.stories.jsx:14]: Yepp, linting

File: src/app/components/ConsentBanner/index.stories.jsx (hunk 14 above)
Comment (dr3) [reply src/app/components/ConsentBanner/index.stories.jsx]: image

Its to simulate the real banner, Ive updated the text to make it clearer

**Review by pjlee11 — DISMISSED** [review]

**Review by pjlee11 — APPROVED** [review]

File: src/app/components/ConsentBanner/index.stories.jsx (hunk 15 above)
Comment (ChrisBAshton) [reply src/app/components/ConsentBanner/index.stories.jsx:14]: 🤢

**Review by ChrisBAshton — APPROVED** [review]

**Comment by ChrisBAshton** [comment]
Looks good to me! 👍 tested with VoiceOver too.

**Comment by ChrisBAshton** [comment]
Simorgh-infrastructure failed latest though 😢

**Pull request merged**